src/main/java/com/google/collegeplanner/servlets/CourseListServlet.java

```diff
@@ -44,13 +44,18 @@ public CourseListServlet(ApiUtil apiUtil) {
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO: Store and retrieve from Datastore
-
     // Create the URI and specify the parameters.
+    String department = request.getParameter("department");
+    if (department == null || department == "") {
+      respondWithError(
+          "Invalid query parameters.", HttpServletResponse.SC_INTERNAL_SERVER_ERROR, response);
+      return;
+    }
```

Comment:
Should be error 400 bad request

Reply:
not fixed yet

Reply:
oops sorry accidentally git reset after making this change. Should be good now